src/backend/marsha/core/serializers/video.py

```diff
@@ -251,11 +286,29 @@ class Meta:  # noqa
     )
     shared_live_medias = serializers.SerializerMethodField()
     playlist = PlaylistLiteSerializer(read_only=True)
-    has_transcript = serializers.SerializerMethodField()
     live_info = serializers.SerializerMethodField()
     xmpp = serializers.SerializerMethodField()
     title = serializers.CharField(allow_blank=False, allow_null=False, max_length=255)
 
+    def to_representation(self, instance):
+        """
+        compute the has_transcript field based on existing timed_text_tracks data
+        already present in the serialized data. Doing this help us to decrease the
+        number of queries made against the database. We have all the data we need.
+        """
+        rep = super().to_representation(instance)
+        timed_text_tracks = rep.get("timed_text_tracks")
+        has_transcript = False
+        for timed_text_track in timed_text_tracks:
+            if timed_text_track.get(
+                "mode"
+            ) == TimedTextTrack.TRANSCRIPT and timed_text_track.get(
+                "is_ready_to_show", False
+            ):
+                has_transcript = True
```

Comment:
Maybe we can use `any` here ?
Something like :
```suggestion
    has_transcript = any(
        timed_text_track.get("mode") == TimedTextTrack.TRANSCRIPT and timed_text_track.get("is_ready_to_show", False)
        for timed_text_track in timed_text_tracks
    )
```

Reply:
once linted we have that:

```
rep["has_transcript"] = any(
            timed_text_track.get("mode") == TimedTextTrack.TRANSCRIPT
            and timed_text_track.get("is_ready_to_show", False)
            for timed_text_track in timed_text_tracks
        )
```

WDYT ?

Reply:
<img src="https://media4.giphy.com/media/N7zYBLiyq6MJZCs28W/giphy.gif"/>

Reply:
![](https://media.giphy.com/media/XNX9uw7fykn5e/giphy.gif)

Reply:
![](https://media.giphy.com/media/ePeHKwWSed0Ag/giphy-downsized.gif)

Reply:
<img src="https://media2.giphy.com/media/p8GJOXwSNzQPu/giphy.gif"/>

Reply:
it's less readable IMO but I'm definitely not a pythonista I guess.

Reply:
Feel free to discard my comment.